content/knowledge/consensus_algorithms.md

```diff
@@ -0,0 +1,314 @@
+For determining the prevailing sentiment of the network, this is as good as having a majority.
+Intuitively, it means that if any quorum agrees to statement `X`, no other quorum can ever agree to `not-X`, because it will necessarily include some node from the first quorum that has already voted for X.
+This is the transitive closure property that this type of systems usually use to fight against Sybil attacks.
+
+The overlapping nodes could be all Byzantine nodes.
+In that case, having quorum intersection does not help the network to agree at all.
+For that reason, the most important assumption about quorum intersection model is that the network enjoys quorum intersection `afer` all misbehaving nodes are removed from the network.
+
+A living example of such a network is internet itself. It is comprised of independent nodes that know about few local to them, but the sets overlap with each other and every node is reachabe through any other by routes and gateways.
+In the same manner, FBA network resembles inter-domain network routing with no central authority dictating or arbitraging agreements.
+If we think of quorum slices as analogous to network reachability and quorums as analogous to transitive reachability, then the Internet's nearly a complete transitive reachability suggests it can be likewise ensured a worldwide consesus with FBA.
+
+Nodes can be categorized as either well-behaved or ill-behaved.
+A well-behaved node chooses sensible quorum slices and obeys the protocol, including eventually responding to all requests.
+An ill-behaved nodes does not.
+Ill-behaved nodes suffer Byzantine failure, meaning they can behave arbitrarily.
+Since the blockchains network most iportant goal is to reach consensus deterministically - meaning reach to the same consensus value every time in the same circumstances, nodes which are generally acting arbitrarily (undeterministically) cannot be considered functioning parts of the system.
+Expected behavior of well-behaved nodes is always `deterministic` in the sense that they will chose the same right value among others every time.
+
+The goal of Byzantine agreement is to ensure that well-behaved nodes externalize the same values despite the presence of the ill-behaved nodes.
+The first part of this goal is to prevent nodes from diverging and externalizing different values for the same slot.
+Second part is to ensure the nodes can externalize values practically always, as opposed to getting blocked in some dead-state.
+Corresponding to these properties are notions of safety and liveness.
+
+TODO: add a picture of voting steps
+
+#### Stellar
+
+One of the most famous consensus protocol proposed, that uses Avalanche and FBA to build decentralized networks is [Stellar Consensus Protocol (SCP)]((https://www.stellar.org/papers/stellar-consensus-protocol?locale=en)), that was decribed in 2015.
```

Comment:
decribed -> described

Reply:
Yes, missed this
thanks!

Reply:
Corrected https://github.com/optakt/optakt.github.io/pull/21/commits/f524c8d94d32f6edd6da60a9e7f74a848c44d4ce#diff-4bf8323c6426a93da1ce22be1b8528779ef6b61fdacef3e0d065f1e1dd47bea5R117
Thanks!